salalib/shapemap.h

```diff
@@ -474,11 +475,13 @@ class ShapeMap : public PixelBase
    void setEditable(bool on = true) 
    { m_editable = on; }
 protected:
-   MapInfoData *m_mapinfodata;
+   bool m_hasMapInfoData = false;
+   MapInfoData m_mapinfodata;
 public:
+   bool hasMapInfoData() const {return m_hasMapInfoData;}
    int loadMifMap(istream& miffile, istream& midfile);
-   bool outputMifMap(ostream& miffile, ostream& midfile) const;
-   const MapInfoData *getMapInfoData() const
+   bool outputMifMap(ostream& miffile, ostream& midfile);

```

Comment:
Why did we lose the const qualifier here?

Reply:
Because that function calls MapInfoData::exportFile which in turns calculates (and stores) the bounds of the MapInfoData object if they have not been calculated (MapInfoData.cpp:283)